From 83e8196515ed420bbbb619e015a49a504c56d6bf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Thea=20Sch=C3=B6bl?= <dev@theaninova.de>
Date: Mon, 18 Sep 2023 17:18:35 +0200
Subject: [PATCH 1/1] workaround: led compile error

---
 hid-ftec.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/hid-ftec.h b/hid-ftec.h
index afce2c3..3ddd514 100644
--- a/hid-ftec.h
+++ b/hid-ftec.h
@@ -1,6 +1,9 @@
 #ifndef __HID_FTEC_H
 #define __HID_FTEC_H
 
+// TODO: remove
+#define CONFIG_LEDS_CLASS
+
 #define FANATEC_VENDOR_ID 0x0eb7
 
 #define CLUBSPORT_V2_WHEELBASE_DEVICE_ID 0x0001
-- 
2.41.0

