From 4a1395857cc00fd287926bc281ecd36fde9f45fc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Thea=20Sch=C3=B6bl?= <dev@theaninova.de>
Date: Mon, 18 Sep 2023 17:30:11 +0200
Subject: [PATCH 1/1] fix: nix makefile adjustments

---
 Makefile | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index 0165e72..74b113c 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,5 @@
 KVERSION ?= `uname -r`
 KDIR ?= /lib/modules/${KVERSION}/build
-MODULEDIR ?= /lib/modules/${KVERSION}/kernel/drivers/hid
 
 default:
 	@echo -e "\n::\033[32m Compiling Fanatec kernel module\033[0m"
@@ -12,12 +11,10 @@ clean:
 	@echo "========================================"
 	$(MAKE) -C $(KDIR) M=$$PWD clean
 
-install:
+install: default
 	@echo -e "\n::\033[34m Installing Fanatec kernel module/udev rule\033[0m"
 	@echo "====================================================="
-	@cp -v hid-fanatec.ko ${MODULEDIR}
-	@cp -v fanatec.rules /etc/udev/rules.d/99-fanatec.rules
-	depmod
+	$(MAKE) -C $(KDIR) M=$$PWD INSTALL_MOD_PATH=$$out modules_install
 
 uninstall:
 	@echo -e "\n::\033[34m Uninstalling Fanatec kernel module/udev rule\033[0m"
-- 
2.41.0

